### PR TITLE
Add updates to piechart tooltip description

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- Task: https://github.com/collective-service/ifrc-gates-dashboard/issues/248

## Changes:
- Add changes to the description of pie chart tooltip and also add calculations logic

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/components/PercentageStats/index.tsx [hunk 1]
```diff
@@ -87,7 +84,7 @@ function PercentageStats(props: Props) {
                 </>
             )}
             headingSize={headingSize}
-            headerDescription={headerDescription}
+            headerDescription={headerDescription || subHeading}
```
Comment (AdityaKhatri) [inline app/components/PercentageStats/index.tsx]: Looks sus. @subinasr can you check all the places where PercentageStats is used?

File: app/components/CustomTooltip/index.tsx [hunk 2]
```diff
@@ -28,12 +38,17 @@ function CustomTooltip(props: Props) {
...
     } = props;
 
     const uncertaintyRange = format === 'percent'
         ? ` [${minValue}% - ${maxValue}%]`
         : ` [${formatNumber(format, minValue)} - ${formatNumber(format, maxValue)}]`;
 
+    const calculatedTotal = customTooltipData?.reduce(
+        (acc, obj) => (acc + (obj?.contextIndicatorValue ?? 1)), 0,
+    ) ?? 1;
+
```
Comment (AdityaKhatri) [inline app/components/CustomTooltip/index.tsx]: use useMemo in both

File: app/components/CustomTooltip/index.tsx [hunk 3]
```diff
@@ -47,16 +62,37 @@ function CustomTooltip(props: Props) {
...
+                                'raw',
+                                item.contextIndicatorValue,
+                            )} (${formatNumber(
+                                'percent',
+                                item.contextIndicatorValue,
+                                calculatedTotal,
+                            )})`)}
+                        {(isDefined(minValue) && isDefined(maxValue))
+                            ? uncertaintyRange
+                            : null}
```
Comment (AdityaKhatri) [inline app/components/CustomTooltip/index.tsx]: This should be simplified and defined outside the render.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Dashboard/Country/index.tsx [hunk 4]
```diff
@@ -923,7 +923,7 @@ function Country(props: Props) {
                                 Health Security and was developed with Economist Impact.
                                 Link:
                                 &nbsp;
-                                <a href="https://www.ghsindex.org/">
+                                <a href="https://www.ghsindex.org/" target="_blank" rel="noopener noreferrer">
```
Comment (tnagorra) [inline app/views/Dashboard/Country/index.tsx]: Let's move the props on separate lines

File: app/components/CustomTooltip/index.tsx [hunk 5]
```diff
@@ -28,11 +38,69 @@ function CustomTooltip(props: Props) {
...
+
+    const calculatedTotal = useMemo(() => (
+        customTooltipData?.reduce(
+            (acc, obj) => (acc + (obj?.contextIndicatorValue ?? 1)), 0,
+        ) ?? 1
+    ), [
+        customTooltipData,
+    ]);
+
+    const tooltipRender = useMemo(() => (
```
Comment (tnagorra) [inline app/components/CustomTooltip/index.tsx]: Let's just use if-else statements for readability

File: app/components/CustomTooltip/index.tsx [hunk 6]
```diff
@@ -28,11 +38,69 @@ function CustomTooltip(props: Props) {
...
+        maxValue,
+        format,
+    ]);
+
+    const calculatedTotal = useMemo(() => (
+        customTooltipData?.reduce(
+            (acc, obj) => (acc + (obj?.contextIndicatorValue ?? 1)), 0,
+        ) ?? 1
+    ), [
+        customTooltipData,
```
Comment (tnagorra) [inline app/components/CustomTooltip/index.tsx]: What does this do? Why is the default value `1`? Why not use an existing function?

File: app/components/CustomTooltip/index.tsx [hunk 7]
```diff
@@ -28,11 +38,69 @@ function CustomTooltip(props: Props) {
...
     } = props;
 
-    const uncertaintyRange = format === 'percent'
-        ? ` [${minValue}% - ${maxValue}%]`
-        : ` [${formatNumber(format, minValue)} - ${formatNumber(format, maxValue)}]`;
+    const uncertaintyRange = useMemo(() => (
+        format === 'percent'
+            ? ` [${minValue}% - ${maxValue}%]`
+            : ` [${formatNumber(format, minValue)} - ${formatNumber(format, maxValue)}]`
```
Comment (tnagorra) [inline app/components/CustomTooltip/index.tsx]: It's best if we do not have a space in front and add space where it is used.

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**